Data/Text/ICU/Spoof.hsc

```diff
@@ -0,0 +1,273 @@
+import Foreign.Ptr (Ptr, castPtr, nullPtr, plusPtr)
+import Foreign.ForeignPtr (withForeignPtr)
+
+-- $api
+--
+
+data SpoofCheck = SingleScriptConfusable
+                | MixedScriptConfusable
+                | WholeScriptConfusable
+                | AnyCase
+                | RestrictionLevel
+                | Invisible
+                | CharLimit
+                | MixedNumbers
+                | AllChecks
+                | AuxInfo
+                deriving (Bounded, Enum, Eq, Show)
+
+instance ToBitMask SpoofCheck where
+  toBitMask SingleScriptConfusable = #const USPOOF_SINGLE_SCRIPT_CONFUSABLE
+  toBitMask MixedScriptConfusable = #const USPOOF_MIXED_SCRIPT_CONFUSABLE
+  toBitMask WholeScriptConfusable = #const USPOOF_WHOLE_SCRIPT_CONFUSABLE
+  toBitMask AnyCase = #const USPOOF_ANY_CASE
+  toBitMask RestrictionLevel = #const USPOOF_RESTRICTION_LEVEL
+  toBitMask Invisible = #const USPOOF_INVISIBLE
+  toBitMask CharLimit = #const USPOOF_CHAR_LIMIT
+  toBitMask MixedNumbers = #const USPOOF_MIXED_NUMBERS
+  toBitMask AllChecks = #const USPOOF_ALL_CHECKS
+  toBitMask AuxInfo = #const USPOOF_AUX_INFO
+
+type USpoofCheck = Int32
+
+data RestrictionLevel = ASCII
+                      | SingleScriptRestrictive
+                      | HighlyRestrictive
+                      | ModeratelyRestrictive
+                      | MinimallyRestrictive
+                      | Unrestrictive
+                      deriving (Bounded, Enum, Eq, Show)
+
+instance ToBitMask RestrictionLevel where
+  toBitMask ASCII = #const USPOOF_ASCII
+  toBitMask SingleScriptRestrictive = #const USPOOF_SINGLE_SCRIPT_RESTRICTIVE
+  toBitMask HighlyRestrictive = #const USPOOF_HIGHLY_RESTRICTIVE
+  toBitMask ModeratelyRestrictive = #const USPOOF_MODERATELY_RESTRICTIVE
+  toBitMask MinimallyRestrictive = #const USPOOF_MINIMALLY_RESTRICTIVE
+  toBitMask Unrestrictive = #const USPOOF_UNRESTRICTIVE
+
+type URestrictionLevel = Int32
+
+data SpoofCheckResult = CheckOK
+                      | CheckFailed [SpoofCheck]
+                      | CheckFailedWithRestrictionLevel { checks :: [SpoofCheck],
+                                                           level :: RestrictionLevel }
+                deriving (Eq, Show)
+
+data SkeletonTypeOverride = SkeletonSingleScript
+                          | SkeletonAnyCase
+                          deriving (Bounded, Enum, Eq, Show)
+
+instance ToBitMask SkeletonTypeOverride where
+  toBitMask SkeletonSingleScript = #const USPOOF_SINGLE_SCRIPT_CONFUSABLE
+  toBitMask SkeletonAnyCase = #const USPOOF_ANY_CASE
+
+type USkeletonTypeOverride = Int32
+
+makeSpoofCheckResult :: USpoofCheck -> SpoofCheckResult
+makeSpoofCheckResult c =
+  case spoofChecks of
+    [] -> CheckOK
+    _ ->
+      case restrictionLevel of
+        Nothing -> CheckFailed spoofChecks
+        Just l -> CheckFailedWithRestrictionLevel spoofChecks l
+  where spoofChecks = fromBitMask $ fromIntegral $ c .&. #const USPOOF_ALL_CHECKS
```

Comment:
Just as a style nit, I like to align the `=` signs in these blocks of declarations. Most of the time. (Sometimes I forget, or am gloriously inconsistent.)